### PR TITLE
Add IAM Permissions so nodes can access AWS ECR

### PR DESCRIPTION
**Pull request by KashifSaadat**
Make the ECR IAM Permissions optional in strict mode, via use of the following IAMSpec:
```yaml
 iam:
    legacy: false
    allowContainerRegistry: true
```

### CODE REVIEW

**Comment by KashifSaadat** [comment]
/assign @chrislovecnm 
/assign @justinsb 

Fixes #3665. This permission set was previously available with the legacy IAM permissions but mistakenly missed when I refactored the IAM Builder code. The access list is Get/List privileges so should be fairly harmless and ok to grant to all nodes.

**Comment by KashifSaadat** [comment]
I've made this optional in https://github.com/kubernetes/kops/pull/3690/commits/c8072c2749e3b9d34fe9cfa0f53bc60ce38f5c87.

File: pkg/apis/kops/cluster.go [hunk 1]
```diff
@@ -180,7 +180,8 @@ type Assets struct {
 
 // IAMSpec adds control over the IAM security policies applied to resources
 type IAMSpec struct {
-	Legacy bool `json:"legacy"`
+	Legacy   bool `json:"legacy"`
+	AllowECR bool `json:"allowECR,omitempty"`
```
Comment (chrislovecnm) [inline pkg/apis/kops/cluster.go]: How about allowContainerRegistry? Someone more cloud generic?

Comment (KashifSaadat) [reply pkg/apis/kops/cluster.go]: Sure sounds good, I've just updated this.

**Pull request merged**